### ISSUE TITLE
Support MacOSX.10.12.sdk

### ISSUE THREAD

**Issue by zchee** [tpoechtrager/osxcross#81]
First of all, Thank for the great project!
osxcross is possible to building uses macOS Frameworks on the Linux. very amazing.
I'm using Go(golang) cross compiling on the Docker container. Thanks.

BTW, now osxcross seems does not support Sierra(10.12) sdk. Maybe needs fix `build.sh`(and `gen_sdk_package.sh`?) script(s).
AFAIK, macOS Sierra SDK has been changed bellow.

`/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk`
has been changed symlink to
`/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk`
At result, remove the version suffix to directory name after extract tar.gz
- https://github.com/tpoechtrager/osxcross/blob/master/build.sh#L253-L261

So, I was trying to fix the `build.sh` script for corresponding that change.
This is very ugly hack, but added

``` diff
diff --git a/build.sh b/build.sh
index 361f6a8..6f006e0 100755
--- a/build.sh
+++ b/build.sh
@@ -87,6 +87,7 @@ case $SDK_VERSION in
   10.9*) TARGET=darwin13; X86_64H_SUPPORTED=1; ;;
   10.10*) TARGET=darwin14; X86_64H_SUPPORTED=1; ;;
   10.11*) TARGET=darwin15; X86_64H_SUPPORTED=1; ;;
+  10.12*) TARGET=darwin16; X86_64H_SUPPORTED=1; ;;
   *) echo "Invalid SDK Version" && exit 1 ;;
 esac

@@ -256,6 +257,8 @@ rm -rf $SDK_DIR/MacOSX$SDK_VERSION* 2>/dev/null

 if [ "$(ls -l SDKs/*$SDK_VERSION* 2>/dev/null | wc -l | tr -d ' ')" != "0" ]; then
   mv -f SDKs/*$SDK_VERSION* $SDK_DIR
+elif [ "$(ls -l MacOSX.sdk 2>/dev/null | wc -l | tr -d ' ')" != "0" ]; then
+  mv -f MacOSX.sdk $SDK_DIR/MacOSX$SDK_VERSION.sdk
 else
   mv -f *OSX*$SDK_VERSION*sdk* $SDK_DIR
 fi
```

However, that is failed on unit test. error message is bellow.

``` sh
$ OCDEBUG=1 UNATTENDED=yes MACOSX_DEPLOYMENT_TARGET=10.12 SDK_VERSION=10.12 OSX_VERSION_MIN=10.12 JOBS=8 /osxcross/build.sh
.
.
.
testing o32-clang ... + o32-clang /osxcross/oclang/test.c -O2 -Wall -o test
osxcross: debug: --> o32-clang /osxcross/oclang/test.c -O2 -Wall -o test
osxcross: debug: <-- /usr/lib/llvm-3.8/bin/clang (clang) -target i386-apple-darwin16 -mlinker-version=264.3.102 -isystem /usr/lib/llvm-3.8/bin/../lib/clang/3.8.0/include -isysroot /osxcross/target/bin/../SDK/MacOSX10.12.sdk -cxx-isystem /osxcross/target/bin/../SDK/MacOSX10.12.sdk/usr/include/c++/v1 -mmacosx-version-min=10.12.0 -arch i386 -Wno-liblto /osxcross/oclang/test.c -O2 -Wall -o test
osxcross: debug: === time spent in wrapper: 0.191463 ms
ld: unexpected token: !tapi-tbd-v2 file '/osxcross/target/bin/../SDK/MacOSX10.12.sdk/usr/lib/libSystem.tbd' for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
+ _exit
+ EC=1
+ '[' 1 -ne 0 ']'
+ test -z build.sh
+ echo ''

exiting with abnormal exit code (1)
+ echo 'exiting with abnormal exit code (1)'
+ test -n 1
+ declare -f -F remove_locks
+ echo 'removing stale locks...'
+ remove_locks
+ rm -rf /osxcross/build/have_cctools_db1f8d9_darwin16
removing stale locks...
+ echo 'if it is happening the first time, then just re-run the script'
+ echo ''
+ test build.sh = build.sh
+ check_for_bug_1242300
+ '[' -e /etc/issue ']'
if it is happening the first time, then just re-run the script

++ grep -i ubuntu.13.10 /etc/issue
+ '[' '' ']'
```

I have not `bash` experience, so can't send pull request...
If you have time, could you consider a fix for  support macOS Sierra 10.12?
If you want to more error message or any information, I can post this. Please feel free to point out.
# 

BTW, I wrote `Dockerfile` for osxcross. It might be helpful for unit test.
It based on `ubuntu:xenial` small image, installed llvm-3.8-dev, clang-3.8 or etc...
If you use the Docker(or any linux container tools), Can I send a pull request?

**Comment by tpoechtrager** [comment]
>  have not bash experience, so can't send pull request...
If you have time, could you consider a fix for support macOS Sierra 10.12?
If you want to more error message or any information, I can post this. Please feel free to point out.

The current open source linker doesn't support the new *.tdb stubs. I will apply your patches once we have an appropriate linker.

> BTW, I wrote Dockerfile for osxcross. It might be helpful for unit test.
It based on ubuntu:xenial small image, installed llvm-3.8-dev, clang-3.8 or etc...
If you use the Docker(or any linux container tools), Can I send a pull request?

I do not use Docker, but you are always more than welcome to send in a pull request.

**Comment by kunxi** [comment]
I encountered a similar issue with homebrew, and find this workaround:
https://github.com/Homebrew/homebrew-core/issues/5330

After reset the command line with `sudo xcode-select -r`, I can link the binary.
